geometry/polygon_area.cpp

```diff
@@ -0,0 +1,113 @@
+ * the polygon.
+ *
+ * @author [Karimi Zayan](https://github.com/waffleSheep)
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for IO operations
+#include <numeric>   /// for inner_product function
+#include <vector>    /// for vector data structure
+
+/**
+ *  @namespace geometry
+ *  @brief Geometry algorithms
+ */
+namespace geometry {
+/**
+ *  @namespace polygon_area
+ *  @brief Functions for [Polygon
+ * Area](https://en.wikipedia.org/wiki/https://en.wikipedia.org/wiki/Polygon)
+ * implementation.
+ */
+namespace polygon_area {
+/**
+ * Point Struct
+ * @tparam for numeric type of the point
+ */
+template <typename T>
+struct Point {
+    T x, y;  /// `x` and `y` values for a point
+};
+
+/**
+ * Class for Polygon Area
+ * @tparam for the numeric type of the points
+ */
+template <typename T>
+class Polygon_Area {
+    std::vector<Point<T>> points;  /// list of all points in the 2d space
+ public:
+    /**
+     * Constructor of given class
+     *
+     * @param point_list list of all the points in the 2d space
+     */
+    explicit Polygon_Area(std::vector<Point<T>> point_list)
+        : points(std::move(point_list)){};
+    /**
+     * Calculates area of the simple polygon
+     * with the [Shoelace
+     * formula](https://en.wikipedia.org/wiki/Shoelace_formula)
+     *
+     * @returns the area of the simple polygon
+     */
+    T get_polygon_area() {
+        /**
+         * This lambda multiplies the x value of the
+         * first point by the y value of the second point
+         */
+        auto multiply_xy_of_two_points = [](const Point<T> &first_point,
+                                            const Point<T> &second_point) {
+            return first_point.x * second_point.y;
+        };
+        auto first_point = points.begin();             // first point iterator
+        auto last_point = std::prev(points.end(), 1);  // last point iterator
+        auto first_term =
+            std::inner_product(points.begin(), std::prev(points.end(), 1),
+                               std::next(points.begin(), 1), 0, std::plus<T>(),
+                               multiply_xy_of_two_points);
+        auto second_term = std::inner_product(
+            std::next(points.begin(), 1), points.end(), points.begin(), 0,
+            std::plus<T>(), multiply_xy_of_two_points);
+        return 0.5 * std::abs(first_term - second_term +
+                              last_point->x * first_point->y -
+                              first_point->x * last_point->y);
+    }
+};
+}  // namespace polygon_area
+}  // namespace geometry
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() {
+    // Area of polygon enclosed by points (0,0), (0,1), (1,1), (2,0). Expected
+    // value is 1.5
+    std::vector<geometry::polygon_area::Point<double>> v = {
+        {0.0, 0.0}, {0.0, 1.0}, {1.0, 1.0}, {2.0, 0.0}};
+    geometry::polygon_area::Polygon_Area<double> p(v);
+    assert(p.get_polygon_area() == 1.5);
```

Comment:
Please create a separate function for tests, and call that function via the `main` function.